### DISCUSSION TITLE
Add a finally option to `run-commands` executor to execute commands after task completion or interrupt

### DISCUSSION THREAD

**Discussion by JoA-MoS** [nrwl/nx#21811] [Feature Requests]
- [x] I'd be willing to implement this feature ([contributing guide](https://github.com/nrwl/nx/blob/master/CONTRIBUTING.md))

## Description

I would like to add a `finally` or `sigint` handler to the run-commands executor that would allow the user to run a set of commands when the task finishes or when it is interrupted. 

## Motivation

The current use case that I have is for a `docker-serve` command serves a docker container and when the user presses ctrl+c to end serving it leaves the container running, and we have to then manually stop / remove it (or have a cleanup task that runs before running `docker-serve`. But I could see this having broader uses than this one usecase.

## Suggested Implementation

We add a `finally` property to the `RunCommandsOptions` interface and the finally uses the same `RunCommandsOptions` interface and is basically a run-commands executor that is run when all tasks are complete or if the executor is interupted.

Example Target Configuration:
```
{
        "docker-serve": {
            "executor": "nx:run-commands",
            "options": {
                "commands": [
                    "docker run  --env-file apps/my-app/.env.docker-serve -p ${PORT}:${PORT} --name=my-app my-app"
                ],
                "parallel": false
                "finally": {
                    "commands": [
                       "docker ps -aq --filter=\"name=my-app\" | xargs -r docker stop",
                       "docker ps -aq --filter=\"name=my-app\" | xargs -r docker rm"
                      ],
                     "parallel": false
                 }
            },
            "dependsOn": ["docker-build"]
        }
}
```

## Alternate Implementations

I am not sure but I am open to suggestions

**Comment by mattvgm** [comment]
I Also think this is a great idea, makes sense. I think in older versions the SIGINT was propagated to the child process. Maybe it was a bug they fixed, but makes a lot of sense to have it.
